Added osu irc client class

Gr3g027 opened the pull request:
The IRC works, whats left to do is picking up the osu link from a chat message and send it.

Commit:
Added osu irc client class

Sclafus commented on gosumemoryClient.cpp:
```diff
@@ -34,15 +34,15 @@ void GosumemoryClient::onConnected()
 {
 	if (readEnabled){
-		qDebug() << "GosumemoryClient: message received from: " << url.toString() << message;
+        // qDebug() << "GosumemoryClient: message received from: " << url.toString() << message;
```
Watch out for whitespaces

Sclafus commented on gosumemoryClient.h:
```diff
@@ -28,7 +28,7 @@ class GosumemoryClient : public QObject {
 		void disconnected();
 
-	private slots:
+    public slots:
```
My bad, thanks for the correction
But watch out for whitespaces 👀

Sclafus commented on osuircclient.cpp:
```diff
@@ -0,0 +1,34 @@
+    ) : QObject(parent), nickname(nickname), password(password), server(server), port(port)
+{
+    // socket = new QTcpSocket(this);
+    connect(&socket, &QTcpSocket::connected, this, &OsuIrcClient::onConnected);
```
This fails the build
I don't know why

Gr3g027 replied:
It was the pointer variable socket, corrected

Sclafus reviewed (changes requested):
Watch out for whitespaces, everything else looks good to me

Commit:
- Irc's socket is not a pointer anymore
- Added sendMsg function in OsuIrcClient class
- All white spaces should be tabs now

Sclafus reviewed (approved):
Looks good to me
